File: 11st/pro_7.java
import java.util.Stack;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> s= new Stack<>();
        for(int i=0; i<moves.length; i++){
        int j=0;
        while(j<board.length&&board[j][moves[i]-1]==0){
            j++;
        }
        if(j<board.length){
            if(!s.empty()&&s.peek()==board[j][moves[i]-1]){
                s.pop(); answer+=2;
            }
            else    s.push(board[j][moves[i]-1]);
            board[j][moves[i]-1]=0;
        }
    }
        return answer;
    }
}

File: 14th/pro_7.cpp
#include <string>
using namespace std;

int solution(string dirs) {
    int answer = 0;
    int ud[11][11]={0,};//음수부분은 6-10
    int lr[11][11]={0,};
    int x=0, y=0;
    for(int i=0; i<dirs.size(); i++){
        if(dirs[i]=='L'){
            if(x<=-5) continue;
            if(!lr[x>=0?x:x+11][y>=0?y:y+11]) answer++;
            lr[x>=0?x:x+11][y>=0?y:y+11]=1;
            x--;
        }
        if(dirs[i]=='R'){
            if(x>=5) continue;
            x++;
            if(!lr[x>=0?x:x+11][y>=0?y:y+11]) answer++;
            lr[x>=0?x:x+11][y>=0?y:y+11]=1;
        }
        if(dirs[i]=='U'){
            if(y>=5) continue;
            y++;
            if(!ud[x>=0?x:x+11][y>=0?y:y+11]) answer++;
            ud[x>=0?x:x+11][y>=0?y:y+11]=1;
        }
        if(dirs[i]=='D'){
            if(y<=-5) continue;
            if(!ud[x>=0?x:x+11][y>=0?y:y+11]) answer++;
            ud[x>=0?x:x+11][y>=0?y:y+11]=1;
            y--;
        }
    }
    return answer;
}